editor/core/files.py

```diff
@@ -124,3 +125,24 @@ def file_from_upload(
         sha256=sha256,
         df=df,
     )
+
+
+def file_from_form(
+    file_type: FileType, type: str, name, description, sha256: str, names: set[str]
+) -> FileObject | FileSet:
+    """Creates a file based on manually added fields."""
+    if type == "File object":
```

Comment:
Is there a way to not use strings here?

Reply:
Done, replaced with constants but I can clean this up a bit more in a dedicated PR